src/aquarium-optimized/Context.h

```diff
@@ -21,41 +25,56 @@ class Model;
 enum BACKENDTYPE : short;
 enum MODELGROUP : short;
 enum MODELNAME : short;
+enum TOGGLE : short;
+
 struct Global;
 
 class Context
 {
   public:
-    Context(){}
-    virtual bool createContext(BACKENDTYPE backend, bool enableMSAA) = 0;
+    Context() {}
     virtual ~Context() {}
-    virtual Texture *createTexture(std::string name, std::string url)                      = 0;
-    virtual Texture *createTexture(std::string name, const std::vector<std::string> &urls) = 0;
+    virtual bool initialize(
+        BACKENDTYPE backend,
+        const std::bitset<static_cast<size_t>(TOGGLE::TOGGLEMAX)> &toggleBitset)              = 0;
+    virtual Texture *createTexture(std::string name, std::string url)                         = 0;
+    virtual Texture *createTexture(std::string name, const std::vector<std::string> &urls)    = 0;
     virtual Buffer *createBuffer(int numComponents, std::vector<float> &buffer, bool isIndex) = 0;
     virtual Buffer *createBuffer(int numComponents,
                                  std::vector<unsigned short> &buffer,
                                  bool isIndex)                                                = 0;
-    virtual Program *createProgram(std::string vId, std::string fId)                       = 0;
-    virtual void setWindowTitle(const std::string &text)                                   = 0;
-    virtual bool ShouldQuit()                                                              = 0;
-    virtual void KeyBoardQuit()                                                            = 0;
-    virtual void DoFlush()                                                                 = 0;
-    virtual void Terminate()                                                               = 0;
+    virtual Program *createProgram(std::string vId, std::string fId)                          = 0;
+    virtual void setWindowTitle(const std::string &text)                                      = 0;
+    virtual bool ShouldQuit()                                                                 = 0;
+    virtual void KeyBoardQuit()                                                               = 0;
+    virtual void DoFlush()                                                                    = 0;
+    virtual void Terminate()                                                                  = 0;
     virtual void FlushInit() {}
-    virtual void preFrame() = 0;
+    virtual void preFrame()   = 0;
     virtual void showWindow() = 0;
 
     int getClientWidth() const { return mClientWidth; }
     int getclientHeight() const { return mClientHeight; }
+    std::bitset<static_cast<size_t>(TOGGLE::TOGGLEMAX)> getDefaultToggleBitset()
```

Comment:
It could be a bit misleading to call it "default toggle" here. Maybe a better name could be "AvaliableToggle"?